broadinstitute/scp-ingest-pipeline
# Prototype DE pipeline for dense matrix inputs (SCP-4074)

To allow progress on engineering-side development work for the differential expression (DE) feature, this PR aims to provide a working "happy path" DE process that generates output files from a DE analysis on a couple of known good datasets (SCP1671 and SCP1539; Note, the test case is derived from SCP1677 - the cell_type assignments in the de_integration_metadata.tsv file are bogus)

This prototype pipeline still needs tests (both integration and unit tests), try/except clauses etc. and capability to run on studies that have sparse matrix. Suggestions welcomed for how best to harden this code - please indicate whether the suggestion should be implemented for this PR or can be incorporated into a downstream "hardening" PR. 

To test:

activate the scp-ingest-pipeline repo virtualenv (added scanpy, changed )
then from the ingest directory of the scp-ingest-pipeline repo:

`source ../scripts/setup_mongo_dev.sh <path to your Github token file>`
`unset BARD_HOST_URL`

`python ingest_pipeline.py --study-id addedfeed000000000000000 --study-file-id dec0dedfeed1111111111111 differential_expression --annotation cell_type__ontology_label --matrix-file-path ../tests/data/differential_expression/de_integration.tsv --matrix-file-type dense --cell-metadata-file ../tests/data/differential_expression/de_integration_unordered_metadata.tsv --cluster-file ../tests/data/differential_expression/de_integration_cluster.tsv --name de_integration --study-accession SCPdev --differential_expression `

confirm that the script runs successfully and generates the following files:

> de_integration--cell_type__ontology_label--cholinergic_neuron--wilcoxon.tsv
> de_integration--cell_type__ontology_label--cranial_somatomotor_neuron--wilcoxon.tsv
> de_integration--cell_type__ontology_label--pyramidal_neuron--wilcoxon.tsv
> de_integration--cell_type__ontology_label--somatomotor_neuron--wilcoxon.tsv
> de_integration--cell_type__ontology_label--sympathetic_cholinergic_neuron--wilcoxon.tsv
> de_log.txt

Content of: 

> de_integration--cell_type__ontology_label--cholinergic_neuron--wilcoxon.tsv

can be compared to a reference file at ../tests/data/differential_expression/reference

This PR supports SCP-4074.
